paddle/fluid/operators/detection/mine_hard_examples_op.cc

```diff
@@ -165,85 +165,107 @@ class MineHardExamplesOp : public framework::OperatorWithKernel {
 
  protected:
   void InferShape(framework::InferShapeContext* ctx) const override {
-    PADDLE_ENFORCE(ctx->HasInput("ClsLoss"),
-                   "Input(ClsLoss) of MineHardExamplesOp should not be null.");
-    PADDLE_ENFORCE(
-        ctx->HasInput("MatchIndices"),
-        "Input(MatchIndices) of MineHardExamplesOp should not be null.");
-    PADDLE_ENFORCE(
-        ctx->HasInput("MatchDist"),
-        "Input(MatchDist) of MineHardExamplesOp should not be null.");
-    PADDLE_ENFORCE(
-        ctx->HasOutput("NegIndices"),
-        "Output(NegIndices) of MineHardExamplesOp should not be null.");
-    PADDLE_ENFORCE(ctx->HasOutput("UpdatedMatchIndices"),
-                   "Output(UpdatedMatchIndices) of MineHardExamplesOp should "
-                   "not be null.");
+    PADDLE_ENFORCE_EQ(
+        ctx->HasInput("ClsLoss"), true,
+        platform::errors::NotFound(
+            "Input(ClsLoss) of MineHardExamplesOp should not be null."));
+    PADDLE_ENFORCE_EQ(
+        ctx->HasInput("MatchIndices"), true,
+        platform::errors::NotFound(
+            "Input(MatchIndices) of MineHardExamplesOp should not be null."));
+    PADDLE_ENFORCE_EQ(
+        ctx->HasInput("MatchDist"), true,
+        platform::errors::NotFound(
+            "Input(MatchDist) of MineHardExamplesOp should not be null."));
+    PADDLE_ENFORCE_EQ(
+        ctx->HasOutput("NegIndices"), true,
+        platform::errors::NotFound(
+            "Output(NegIndices) of MineHardExamplesOp should not be null."));
+    PADDLE_ENFORCE_EQ(
+        ctx->HasOutput("UpdatedMatchIndices"), true,
+        platform::errors::NotFound(
+            "Output(UpdatedMatchIndices) of MineHardExamplesOp should "
+            "not be null."));
 
     auto cls_loss_dims = ctx->GetInputDim("ClsLoss");
     auto idx_dims = ctx->GetInputDim("MatchIndices");
     auto dis_dims = ctx->GetInputDim("MatchDist");
 
-    PADDLE_ENFORCE_EQ(cls_loss_dims.size(), 2UL,
-                      "The shape of ClsLoss is [N, Np].");
+    PADDLE_ENFORCE_EQ(
+        cls_loss_dims.size(), 2UL,
+        platform::errors::InvalidArgument("The shape of ClsLoss is [N, Np]."));
```

Comment:
need to print cls_loss_dims.size() according to the request

Reply:
done, thanks.